xarray/core/dataarray.py

```diff
@@ -1369,6 +1369,61 @@ def real(self):
     @property
     def imag(self):
         return self._replace(self.variable.imag)
+    
+    def dot( self, other):
+        """Perform sum product of two DataArrays along their shared dims. 
+        Equivalent to taking taking tensor dot over all shared dims'''
+
+        Parameters
+        ----------
+        other: the DataArray with which the DataArray calling this function 
+        will perform dot with.
+
+        Returns
+        -------
+        DataArray(new_data, new_coords, new_dims) : DataArray
+            new_data: dot performed over shared DataArrays dims
+            new_dims: the dims that were not shared between DataArrays and 
+                      thus were not summed over.
+            new_coords: coords corresponding to those original unique dims.
+            
+        See also
+        --------
+        np.tensordot(a,b, axes)
+
+        Examples
+        --------
+        
+        >>> da.dims
+        ('x_trans', 'y_trans', 'imgID')  
+        >>> dm.dims
+        ('models', 'y_trans_m', 'imgID')
+   
+        >>> t = da.dot( dm )
+        >>> t.dims
+        ('x_trans', 'y_trans', 'models', 'y_trans_m')
+
+        """
+        s = self
```

Comment:
I would probably just still with `self` here? not sure what we gain from going to one letter...